platform/default/online_file_source.cpp

```diff
@@ -152,30 +152,34 @@ OnlineFileSource::~OnlineFileSource() = default;
 
 std::unique_ptr<AsyncRequest> OnlineFileSource::request(const Resource& resource, Callback callback) {
     Resource res = resource;
+    
+    if(apiBaseURL.empty()){
```

Comment:
Set the default value to `mbgl::util::API_BASE_URL` instead of checking for the empty string.